manifests/master/config.pp

```diff
@@ -90,35 +90,35 @@
+  Variant[Stdlib::Port::User, Pattern[/\A[0-9]+\Z/]] $puppetdb_port                  = defined(Class['puppetdb']) ? {
     true    => $puppetdb::disable_ssl ? {
       true => 8080,
       default => 8081,
     },
     default => 8081,
   },
-  $puppetdb_disable_ssl        = defined(Class['puppetdb']) ? {
+  Boolean                                            $puppetdb_disable_ssl           = defined(Class['puppetdb']) ? {
     true    => $puppetdb::disable_ssl,
     default => false,
   },
-  $masterless                  = $puppetdb::params::masterless,
-  $puppetdb_soft_write_failure = false,
-  $manage_routes               = true,
-  $manage_storeconfigs         = true,
-  $enable_storeconfigs         = true,
-  $manage_report_processor     = false,
-  $manage_config               = true,
-  $create_puppet_service_resource = true,
-  $strict_validation           = true,
-  $enable_reports              = false,
-  $puppet_confdir              = $puppetdb::params::puppet_confdir,
-  $puppet_conf                 = $puppetdb::params::puppet_conf,
-  $terminus_package            = $puppetdb::params::terminus_package,
-  $puppet_service_name         = $puppetdb::params::puppet_service_name,
-  $puppetdb_startup_timeout    = $puppetdb::params::puppetdb_startup_timeout,
-  $test_url                    = $puppetdb::params::test_url,
-  $restart_puppet              = true,
+  Boolean                                            $masterless                     = $puppetdb::params::masterless,
+  Boolean                                            $puppetdb_soft_write_failure    = false,
+  Boolean                                            $manage_routes                  = true,
+  Boolean                                            $manage_storeconfigs            = true,
+  Boolean                                            $enable_storeconfigs            = true,
+  Boolean                                            $manage_report_processor        = false,
+  Boolean                                            $manage_config                  = true,
+  Boolean                                            $create_puppet_service_resource = true,
+  Boolean                                            $strict_validation              = true,
+  Boolean                                            $enable_reports                 = false,
+  Stdlib::Absolutepath                               $puppet_confdir                 = $puppetdb::params::puppet_confdir,
+  Stdlib::Absolutepath                               $puppet_conf                    = $puppetdb::params::puppet_conf,
+  String[1]                                          $terminus_package               = $puppetdb::params::terminus_package,
+  String[1]                                          $puppet_service_name            = $puppetdb::params::puppet_service_name,
+  Integer                                            $puppetdb_startup_timeout       = $puppetdb::params::puppetdb_startup_timeout,
```

Comment:
A negative value does not make sense.  We can also set a lower bound, but it is admittedly less important than for a String where legacy code used to use `''` for "no value":

```suggestion
  Integer[0]                                         $puppetdb_startup_timeout       = $puppetdb::params::puppetdb_startup_timeout,
```

If you decide to adjust these, there are more but I did not added comments to each of them to avoid flooding with these less important changes.

Reply:
Each integer has had lower bounds set.